src/engine/GameWorker.ts

```diff
@@ -9,6 +9,8 @@ import { GameResourceManager } from "./resource/GameResourceManager";
 
 const workerScope = globalThis as typeof globalThis & Worker;
 
+const isFirefox = navigator.userAgent.toLowerCase().includes("firefox");
```

Comment:
For user agent check, use the rendering engine instead of the browser:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#rendering_engine

I'd use `Gecko/` as the search string.